### DISCUSSION TITLE
Construction timelapse with external timer

### DISCUSSION THREAD

**Discussion by Orsbo-byte** [gopro/labs#188] [General]
Hi all!

I think GoPro Labs is a great way to extend and configure the functionality of the cameras. Thanks to everyone who is contributing to the development! Im doing long term construction timelapse and I figured a way to avoid random camera freeze over longer timelapse projects was to use an external timer / remote controlled relay on the AC power supply in combination with a boot script. In this way I hope the camera will be reset and restarted every day. For the next  week or so I will be testing this setup with the below script:

!MNAKE=1!MWAKE=1!MBOOT="!Ldaily"!SAVEdaily=mTP>06:05<20:00!1SQ!20:00E!1U~!1U

Timer goes on each day at 06:00 and off at 22:00

This means that if the timer malfunctions and is left on, the capture will stop after one day. I'm a little uncertain how looping and the conditional statement works togehter: Would something like the below version work better?

!MNAKE=1!MWAKE=1!MBOOT="!Ldaily"!SAVEdaily=mTP>06:05<20:00!1SQ!20:00E!1U!1R~!1U!900NQ!1R

Happy to hear your thoghts on how to improve this setup.

cheers /

Martin

**Comment by Orsbo-byte** [comment]
Sorry, got the timing wrong. Should be like this:

!MNAKE=1!MWAKE=1!MBOOT="!Ldaily"!SAVEdaily=mTP>06:00<20:00!1SQ!20:00E!1U~!1U

Timer goes on each day at 06:05 and off at 22:00

**Comment by dnewman-gpsw** [comment]
External timers with boot scripts are very reliable. Nice!

The command is tad malformed.  !SAVEdaily=mTP>06:00<20:00!1S+!20:00E+!U  is cleaner.

**Reply by Orsbo-byte** [reply]
Great! Thanks for the tip, the + signs really makes it more readable. ! 

cheers / Martin

**Reply by kelp-videos** [reply]
Thank you for all your trouble-shooting in these discussion posts. I was wondering what apparatus you used for an external timer and how you attached it to the gopro?

**Comment by jxqqk** [comment]
Thank you for the tips, the external timer with boot script is a very strong solution, resetting the camera every day is a big plus if something goes wrong (ie. thermal problems).
JM

**Comment by amatsumedia** [comment]
Would you be willing to give or make a full extensive tutorial on how to accomplish this?! Super interested!
